### ISSUE TITLE
unable to easily use `ignore_malformed` in data streams

### ISSUE THREAD

**Issue by pmuellr** [elastic/elasticsearch#98511]
### Elasticsearch Version

8.10.0

### Installed Plugins

running from Kibana `yarn es snapshot`

### Java Version

_bundled_

### OS Version

Mac OSX latest

### Problem Description

We’re having difficulties using `ignore_malformed` at the index level in a data stream, as it seems like it really doesn’t want `@timestamp` to have that setting.  This seems like an ES bug, since [the doc says](https://www.elastic.co/guide/en/elasticsearch/reference/current/ignore-malformed.html#ignore-malformed-setting) 
> “The index.mapping.ignore_malformed setting can be set on the index level to ignore malformed content globally across all allowed mapping types. Mapping types that don’t support the setting will ignore it if set on the index level.“.  

Technically, `@timestamp` is date, so not ignored because date types are “allowed”, but you can understand why @timestamp should not use that setting for data streams.

You also can't do the index-level and override just `@timestamp`, as it will complain that you can't use `ignore_malformed` on that field at all. So, in order for us to do what we want, we would have to add ignore_malformed: true on every field except @timestamp, which seems over-the-top hard.

We have the following issue open, blocked on resolution to this: https://github.com/elastic/kibana/issues/161465

### Steps to Reproduce

<details>
<summary>repro in Dev Console</summary>

```
PUT _index_template/pmuellr-index-template
{
  "index_patterns": ["pmuellr-data-stream"],
  "data_stream": { },
  "template": {
    "settings": {
      "index.mapping.ignore_malformed": true 
    },
    "mappings": {
      "properties": {
        "@timestamp": {
          "type": "date"
        },
        "message": {
          "type": "text"
        }
      }      
    }
  }
}

{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "composable template [pmuellr-index-template] template after composition is invalid"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "composable template [pmuellr-index-template] template after composition is invalid",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "invalid composite mappings for [pmuellr-index-template]",
      "caused_by": {
        "type": "illegal_argument_exception",
        "reason": "data stream timestamp field [@timestamp] has disallowed [ignore_malformed] attribute specified"
      }
    }
  },
  "status": 400
}
```

<details>

**Comment by dakrone** [comment]
@jbaiera you fixed this recently in https://github.com/elastic/elasticsearch/pull/96051 , but it looks like if someone manually specifies the `@timestamp` field in the mappings, then it still throws the exception. Should we force it to be `ignore_malformed: false` even if there's a mapping definition?

@pmuellr as a workaround, if you remove the `@timestamp` mapping (using the default), then your request works. This was changed in #96051 to allow using that.

**Comment by pmuellr** [comment]
Ahhh!  Just noticed the doc

> A [date](https://www.elastic.co/guide/en/elasticsearch/reference/master/date.html) or [date_nanos](https://www.elastic.co/guide/en/elasticsearch/reference/master/date_nanos.html) mapping for the @timestamp field. If you don’t specify a mapping, Elasticsearch maps @timestamp as a date field with default options.

I think that'll be fine for us, as we control all the writing to the index.  

Does feel a bit clumsy, especially since we are juggling some code where we sometimes use data streams, and other times use aliases / indices / ilm - but deleting this field from the mappings for data streams should be straight-forward.

**Comment by felixbarny** [comment]
> You also can't do the index-level and override just @timestamp, as it will complain that you can't use ignore_malformed on that field at all.

That sounds surprising to me. Could you share an example where that's the case? This example does work:

```
PUT _index_template/pmuellr-index-template
{
  "index_patterns": ["pmuellr-data-stream"],
  "data_stream": { },
  "template": {
    "settings": {
      "index.mapping.ignore_malformed": true 
    },
    "mappings": {
      "properties": {
        "@timestamp": {
          "ignore_malformed": false,
          "type": "date"
        },
        "message": {
          "type": "text"
        }
      }      
    }
  }
}
```

**Comment by pmuellr** [comment]
Huh, ya, that works for me as well!  I thought I had tried that, and saw a different but similar error.  I must have just seen the original error again, or something.

Thanks for the lead!

Or perhaps this has been "fixed" since then?  I poked around some of the referenced issues, and found this interesting note in PR https://github.com/elastic/elasticsearch/pull/96081 , which makes it sound like that incantation should **NOT** work! :-). But that's an old PR ...

> Explicitly setting ignore_malformed: true on a data stream timestamp mapping will still cause validation to fail at index template creation time.

**Comment by felixbarny** [comment]
That's right, explicitly setting `ignore_malformed: true` will fail. However, setting `ignore_malformed: false` is valid.
